include/pm/policy.h

```diff
@@ -33,6 +59,121 @@ const struct pm_state_info *pm_policy_next_state(uint8_t cpu, int32_t ticks);
 
 /** @endcond */
 
+#if defined(CONFIG_PM) || defined(__DOXYGEN__)
+/**
+ * @brief Increase a power state lock counter.
+ *
+ * A power state will not be allowed on the first call of
+ * pm_policy_state_lock_get(). Subsequent calls will just increase a reference
+ * count, thus meaning this API can be safely used concurrently. A state will
+ * be allowed again after pm_policy_state_lock_put() is called as many times as
+ * pm_policy_state_lock_get().
+ *
+ * Note that the PM_STATE_ACTIVE state is always allowed, so calling this API
+ * with PM_STATE_ACTIVE will have no effect.
+ *
+ * @param state Power state.
+ *
+ * @see pm_policy_state_lock_put()
+ */
+void pm_policy_state_lock_get(enum pm_state state);
+
+/**
+ * @brief Decrease a power state lock counter.
+ *
+ * @param state Power state.
+ *
+ * @see pm_policy_state_lock_get()
+ */
+void pm_policy_state_lock_put(enum pm_state state);
+
+/**
```

Comment:
Wouldn't be better ?
```
pm_policy_state_lock()
pm_policy_state_unlock()
pm_policy_state_is_locked()
```

Reply:
I also thought the same, but I think `pm_policy_state_lock`/`pm_policy_state_unlock` doesn't emphasize that it's a reference counted API. Calling `pm_policy_state_unlock` doesn't necessarily mean the state is unlocked. The terminology used in this proposal is:

- pm_policy_state_lock_put: release (put) _state lock_
- pm_policy_state_lock_get: obtain (get) _state lock_

put/get terminology is re-used from PM runtime.

Reply:
That is a good point ... lets move with this names then. The one that I really disliked is
`pm_policy_state_lock_is_active` :/ But it is better to keep the namespace consistent.